hook_test.go

```diff
@@ -49,3 +50,26 @@ func TestNoHook(t *testing.T) {
 	}
 	defer coll.Close()
 }
+
+// Test loading elfs with valid and invalid HookMap ABI
+func TestCheckHookMap(t *testing.T) {
```

Comment:
I don't think we need a whole ELF to test this, can we just use `NewMap()` (https://godoc.org/github.com/cilium/ebpf#NewMap) to create a map that doesn't match the ABI?

Reply:
Thanks for suggestion; changes applied.